src/main/java/com/mixpanel/android/viewcrawler/ViewVisitor.java

```diff
@@ -99,12 +100,58 @@ protected String name() {
             return "Property Mutator";
         }
 
-        private final Caller mMutator;
-        private final Caller mAccessor;
+        private final PropertyCaller mMutator;
+        private final PropertyCaller mAccessor;
         private final WeakHashMap<View, Object> mOriginalValues;
         private final Object[] mOriginalValueHolder;
     }
 
+    public static class LayoutSetVisitor extends ViewVisitor {
+        public LayoutSetVisitor(List<Pathfinder.PathElement> path, LayoutCaller mutator) {
+            super(path);
+            mOriginalValues = new WeakHashMap<View, int[]>();
+            mMutator = mutator;
+        }
+
+        @Override
+        public void cleanup() {
+            for (Map.Entry<View, int[]> original:mOriginalValues.entrySet()) {
+                final View changedView = original.getKey();
+                final int[] originalValue = original.getValue();
+                mMutator.applyMethodWithArguments(changedView, originalValue);
+            }
+        }
+
+        @Override
+        public void accumulate(View found) {
+            // currentRules is an array which has rule_index as the index and anchor_id as the value
+            // newRule and currentRule are individual rules which look like {rules_index, anchor_id}
+            final RelativeLayout.LayoutParams currentParams = (RelativeLayout.LayoutParams)found.getLayoutParams();
+            final int[] currentRules = currentParams.getRules();
+            final int[] newRule = mMutator.getArgs();
+            final int rule_index = newRule[LayoutCaller.RULE_INDEX];
+            final int[] currentRule = {rule_index, currentRules[rule_index]};
```

Comment:
When we store these, I'd rather store them as a special object rather than an array with magic indexes. It would be more fun to read `rule.verb` and `rule.anchor`